Google.PowerShell/BigQuery/GcbqTable.cs

```diff
@@ -52,65 +53,48 @@ public class GetGcbqTable : GcbqCmdlet
-        /// (This will be used if both InputObject and Dataset are provided)
-        /// </para>
-        /// </summary>
-        [Parameter(Mandatory = false, ValueFromPipeline = true)]
-        public Dataset InputObject { get; set; }
-
-        /// <summary>
-        /// <para type="description">
-        /// The ID of the dataset that you want to get a descriptor object for.
+        /// The ID of the table that you want to get a descriptor object for.
         /// </para>
         /// </summary>
         [Parameter(Mandatory = false, Position = 0)]
         public string Table { get; set; }
 
         protected override void ProcessRecord()
         {
-            // Select the DatasetID. Favor InputObject if both present. Default if neither present.
-            if (InputObject == null && Dataset == null)
-            {
-                ThrowTerminatingError(new ErrorRecord(
-                    new Exception("No Dataset Specified"),
-                    "No Dataset Specified",
-                    ErrorCategory.InvalidArgument,
-                    InputObject));
-                return;
-            }
-            Dataset = (InputObject != null) ? InputObject.DatasetReference.DatasetId : Dataset;
-
-            // Create and execute request.
             if (Table == null)
             {
-                TablesResource.ListRequest request = new TablesResource.ListRequest(Service, Project, Dataset);
-                var response = request.Execute();
-                if (response != null)
-                {
-                    WriteObject(response, true);
-                }
-                else
+                TablesResource.ListRequest request = Service.Tables.List(Project, Dataset);
+                do
                 {
-                    WriteError(new ErrorRecord(
-                        new Exception("400"),
-                        "Error 400: List request to server failed.",
-                        ErrorCategory.InvalidArgument,
-                        Dataset));
+                    TableList response = request.Execute();
+                    if (response == null)
+                    {
+                        WriteError(new ErrorRecord(
+                            new Exception("The List query returned null instead of a well formed list."),
+                            "Null List Returned", ErrorCategory.ReadError, Dataset));
+                    }
+                    if (response.Tables != null)
+                    {
+                        WriteObject(response.Tables, true);
+                    }
+                    request.PageToken = response.NextPageToken;
                 }
+                while (!Stopping && request.PageToken != null);
             }
             else
             {
-                TablesResource.GetRequest request = new TablesResource.GetRequest(Service, Project, Dataset, Table);
+                TablesResource.GetRequest request = Service.Tables.Get(Project, Dataset, Table);
```

Comment:
Again, adding a function here to match the function above would help readability.